### PR TITLE
Add a view button by default to parent taxon QCBX

### LINKED ISSUE #4895: Add a view button to parent query combo boxes by default
image

With the release of 7.9.4, query combo boxes will no longer give the user the ability to view a parent tree record since the default forms explicitly say to not display the edit button. We should add a `viewbtn` to the `parent` query combo box unless it is explicitly disabled as this is an unnecessary loss in functionality users have come to rely on.

```xml
<cell type="field" id="parent" uitype="querycbx" isrequired="true" colspan="3" initialize="name=Taxon;title=Taxon;editbtn=false;newbtn=false;editoncreate=true" name="parent"/>
```

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #4895

### Checklist

- [x] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [x] Add relevant issue to release milestone

### Testing instructions

- in a taxon data entry form: 
- [ ] Verify that there is a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=false" and no viewBtn=... 
- [ ] Verify that there is a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=false" and viewBtn=true
- [ ] Verify that there is a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=true" and viewBtn=true
- [ ] Verify that there is not a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=true" and no viewBtn=...   
- [ ] Verify that there is not a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=true" and viewBtn=false 

- [ ] can be verified too for other tables and QCBX fields 

<cell type="label" labelfor="parent"/>
<cell type="field" id="parent" uitype="querycbx" isrequired="true" initialize="name=Taxon;title=Taxon;editbtn=true;newbtn=false;editoncreate=true;" name="parent" colspan="3"/>

<cell type="label" labelfor="parent"/>
<cell type="field" id="parent" uitype="querycbx" isrequired="true" initialize="name=Taxon;title=Taxon;editbtn=true;newbtn=false;editoncreate=true;viewBtn=true" name="parent" colspan="3"/>

### CODE REVIEW

File: specifyweb/frontend/js_src/lib/components/FormParse/fields.ts [hunk 1]
```diff
@@ -212,18 +212,23 @@ const processFieldType: {
       maxLength: f.parseInt(getProperty('maxLength')),
     };
   },
-  QueryComboBox({ getProperty, fields }) {
+  QueryComboBox({ getProperty, fields, table, cell }) {
     if (fields === undefined) {
       console.error('Trying to render a query combobox without a field name');
       return { type: 'Blank' };
     } else if (fields.at(-1)?.isRelationship === true) {
+      const isParentTaxon =
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/FormParse/fields.ts:220]: why make this logic specific to a single table?
try to avoid introducing edge cases - they make testing harder as there are more cases needed to test to exercise all the code path (though in practice that more likely means that table-specific or collection-specific bugs could be hidden in the code for longer until some user discovers them)

instead, if the form definition explicitly says `editBtn=false`, but does not explicitly include `viewBtn`, then why not default viewBtn to true in that case?
that logic seems desirable in all tables, not just Taxon...

**Review by maxpatiiuk — CHANGES_REQUESTED** [review]

**Review by melton-jason — APPROVED** [review]
Glad there is a generic solution here!

**Review by combs-a — APPROVED** [review]
### Testing instructions

- in a taxon data entry form: 
- [X] Verify that there is a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=false" and no viewBtn=... 
- [X] Verify that there is a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=false" and viewBtn=true
- [X] Verify that there is a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=true" and viewBtn=true
- [X] Verify that there is not a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=true" and no viewBtn=...   
- [X] Verify that there is not a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=true" and viewBtn=false 

- [X] can be verified too for other tables and QCBX fields 

Everything passed fine, also worked on Cataloger and Accession in Collection Object. Looking good! 🎊 

Parent QCBX in Geography would be a good one to test as well, I checked it briefly and it does have the view button visible but I didn't do anything beyond that.

**Review by alesan99 — APPROVED** [review]
- [x] Verify that there is a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=false" and no viewBtn=... 
- [x] Verify that there is a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=false" and viewBtn=true
- [x] Verify that there is a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=true" and viewBtn=true
- [x] Verify that there is not a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=true" and no viewBtn=...   
- [x] Verify that there is not a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=true" and viewBtn=false 

- [x] can be verified too for other tables and QCBX fields 

Behaves correctly with Taxon and Accession QCBXs 👍

**Review by Areyes42 — APPROVED** [review]
### Testing instructions

-   in a taxon data entry form:
    
-  [x]  Verify that there is a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=false" and no viewBtn=...
    
-  [x]   Verify that there is a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=false" and viewBtn=true
    
-  [x]   Verify that there is a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=true" and viewBtn=true
    
-  [x]   Verify that there is not a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=true" and no viewBtn=...
    
-   [x]  Verify that there is not a view button next to the parent QCBX when the form def has the attribute initialize="editbtn=true" and viewBtn=false
    
-   [x]  can be verified too for other tables and QCBX fields

Looks good! Tested on Taxon and Geography and both worked properly, nice job.

**Review by maxpatiiuk — APPROVED** [review]
